### DISCUSSION TITLE
Do not work

### DISCUSSION THREAD

**Discussion by tekevics** [Arkady23/V380-Web#53] [General]
Hi!

I have a camera, using this 'hack', but have a message:

o.js:81          POST http://IP/cgi-bin/sdcard.sh 404 (Not Found)

Made a printenv:

CPU: ARM926EJ-S [41069265] revision 5 (ARMv5TEJ), cr=0005317f
CPU: VIVT data cache, VIVT instruction cache
Machine model: C330_HS_EVB_CBD_AK3918EV331_PTZ6_0_V1.0.0 dev board
Memory policy: Data cache writeback
ANYKA CPU AK39XXEV330 (ID 0x20160101)

Could somebody help me!

THX in advance

**Comment by Arkady23** [comment]
SD card in fat32? We also need to check the paths. Maybe something was copied really wrong. You'd better not write about the processor, but about the paths. What is the SD structure.  What directories. The card itself should be visible by /mnt/sdcard/

**Reply by tekevics** [reply]
THX for really sudden react! 8-)

1. SD card in fat32?  -> yes

2. What is the SD structure. What directories

ls -al /mnt/sdcard/

drwxr-xr-x    6 root     root          4096 Jan  2 00:31 .
drwxrwxrwx    5 1000     1000            52 Mar 19  2021 ..
drwxr-xr-x    3 root     root          4096 Jan  1 21:41 RecFiles
drwxr-xr-x    2 root     root          4096 Jan  1 00:00 ark-add-on
drwxr-xr-x    2 root     root          4096 Jan  1 00:00 bin
drwxr-xr-x    2 root     root          4096 Jan  1 03:43 cgi-bin
-rwxr-xr-x    1 root     root           854 Sep  5  2022 index.html.gz
-rwxr-xr-x    1 root     root            13 Aug  3  2022 quick_check.ini
-rwxr-xr-x    1 root     root          2113 Jan  1 04:50 quick_check_dump_26098981.txt
-rwxr-xr-x    1 root     root           332 Jan  1 01:08 setup.sh

**Reply by Arkady23** [reply]
It looks like you have not completed the installation. After proper installation, these files should not be:
quick_check.in
quick_check_dump_26098981.txt
setup.sh
Then you need to perform the installation via setup.sh and show the protocol. What messages were issued?

**Reply by tekevics** [reply]
'... you have not completed the installation. ...'
Yes, you're right! I did not understand why it is. But after ran the setup.sh it is the state. But why?

**Comment by Arkady23** [comment]
I don't understand you. Type commands from setup.sh manually via telnet. And watch the messages. Write down what happened and how. Copy everything from the screen.

**Reply by tekevics** [reply]
[root@V380E ~]$ fl=/mnt/mtd/vg_boot.sh
[root@V380E ~]$ echo $fl
/mnt/mtd/vg_boot.sh
[root@V380E ~]$

**Reply by Arkady23** [reply]
And why did you introduce it? I didn't understand.

**Reply by tekevics** [reply]
sorry, i disunderstand you

**Comment by tekevics** [comment]
if grep -q startup.sh $fl
then
  exit 1
else
...
fi


naturally it is exited

**Reply by Arkady23** [reply]
Enter >setup.sh

**Reply by tekevics** [reply]
Entered, breaked the telnet connection

**Reply by Arkady23** [reply]
That's how it should be. After installation, telnet is configured via the web.

**Reply by Arkady23** [reply]
Open the file.
ark-add-on\startup.sh 
Server launches are regulated there by comments. And telnet too.

**Reply by tekevics** [reply]
THX, sorry i have some time to react. Till worked remotely and lost the connection. 'i'll be back' 8-)

**Reply by Arkady23** [reply]
You can enable ftp, telnet, http.

**Reply by tekevics** [reply]
everything is started, but 

![image](https://user-images.githubusercontent.com/1470490/210327199-b541e562-4ba8-4c56-a64e-d484e8ad6e1a.png)

**Reply by Arkady23** [reply]
I don't know why. I got a new camera for the New Year. I haven't turned it on yet. As soon as my hands reach it, I'll try. Maybe the Chinese have done something new with Linux. Don't know. I'll write you how I can do it.

**Reply by tekevics** [reply]
THX in advance 8-)

**Reply by Arkady23** [reply]
Maybe you have a bad sd card?

**Reply by tekevics** [reply]
i really thought it in after 10000 try 8-)

**Reply by tekevics** [reply]
i do not want to bother you, but the changed sd card is not successed

**Reply by Arkady23** [reply]
I took a new camera with software 2021. I connected it through the application, prepared the sd, formatted it in exFAT under Windows. I logged into telnet and launched setup.sh -- file not found. Really need to enter like this (with an oblique and a dot):
```
[root@V380E /mnt/sdcard]$ ls -l
total 768
drwxr-xr-x    2 root     root        131072 Jan  6 12:40 RecFiles
drwxr-xr-x    2 root     root        131072 Jan  6  2023 System Volume Information
drwxr-xr-x    2 root     root        131072 Aug 14 11:10 ark-add-on
-rwxr-xr-x    1 root     root            13 Aug  3 23:26 quick_check.ini
-rwxr-xr-x    1 root     root          1555 Jan  1  1970 quick_check_dump_57694865.txt
-rwxr-xr-x    1 root     root           319 Sep 24 12:57 setup.sh
[root@V380E /mnt/sdcard]$ ./setup.sh
killall: as9nvserver: no process killed
killall: udp_broadcast: no process killed
```
![image](https://user-images.githubusercontent.com/79393209/211018913-bc69c154-672a-447c-84e0-e2a0dc4b6271.png)

**Comment by h200101** [comment]
Hi! I am having the same problem. I did everything I was supposed to. run `./setup.sh` it reboots, http server boots up.
Processes running on the camera:
```
PID   USER     TIME  COMMAND
    1 root      0:00 init
    2 root      0:00 [kthreadd]
    3 root      0:00 [ksoftirqd/0]
    5 root      0:00 [kworker/0:0H]
    6 root      0:01 [kworker/u2:0]
    7 root      0:00 [kdevtmpfs]
    8 root      0:00 [kworker/u2:1]
   84 root      0:00 [writeback]
   86 root      0:00 [crypto]
   87 root      0:00 [bioset]
   89 root      0:00 [kblockd]
  125 root      0:00 [kswapd0]
  176 root      0:00 [fsnotify_mark]
  209 root      0:00 [bioset]
  212 root      0:00 [bioset]
  215 root      0:00 [bioset]
  218 root      0:00 [bioset]
  221 root      0:00 [bioset]
  224 root      0:00 [bioset]
  227 root      0:00 [bioset]
  230 root      0:00 [bioset]
  238 root      0:00 [spi0]
  242 root      0:00 [bioset]
  247 root      0:00 [bioset]
  252 root      0:00 [bioset]
  257 root      0:00 [bioset]
  262 root      0:00 [bioset]
  267 root      0:00 [bioset]
  272 root      0:00 [bioset]
  277 root      0:00 [bioset]
  282 root      0:00 [bioset]
  287 root      0:00 [bioset]
  304 root      0:00 [deferwq]
  305 root      0:08 [kworker/0:2]
  306 root      0:00 [kworker/0:1H]
  320 root      0:01 [jffs2_gcd_mtd9]
  325 root      0:00 /sbin/getty -L ttySAK0 115200 vt100
  326 root      0:01 syslogd -D -n -O /var/log/messages -s 200 -b 3
  327 root      0:02 klogd -n
  358 root      0:00 [dma_reserved_he]
  389 root      0:00 [bioset]
  390 root      0:00 [mmcqd/0]
  405 root      0:00 [cfg80211]
  416 root      0:00 [usb_otg_wq]
  506 root      0:00 /mvs/apps/mv_watchdog
  507 root      0:00 /mvs/apps/eventhub_core
  509 root      0:03 {eventhub_core_r} /mvs/apps/eventhub_core
  515 root      0:00 /mvs/apps/vsipbroadcast
  517 root      0:00 /mvs/apps/as9updatednsip
  526 root      0:00 /mvs/apps/recorder
  528 root      0:00 /mvs/apps/log_upload
  534 root      0:01 telnetd
  543 root      3:32 {recorder_r} /mvs/apps/recorder
  552 root      0:00 /mnt/sdcard/bin/busybox tcpsvd -vE 0.0.0.0 21 ftpd -w
  555 root      0:00 /mnt/sdcard/bin/busybox httpd -p 80 -h /mnt/sdcard
  573 root      0:00 [devmem]
  617 root      0:00 [phy0-atbm_wq]
  618 root      0:00 [phy0-usb_atbm_b]
  643 root      0:00 wpa_supplicant -B -iwlan0 -Dwext -c /mnt/mtd/wificonf/wpa_supplicant.conf
  688 root      0:00 udhcpc -i wlan0 -r 192.168.178.26
  691 root      0:00 {updatednsip_r} /mvs/apps/as9updatednsip
  724 root      0:00 /bin/login
  728 root      0:00 -sh
  921 root      0:00 [kworker/0:0]
 1108 root      0:00 [kworker/u2:2]
 1116 root      0:00 ps
```

Interestingly, if I run `/mnt/sdcard/bin/busybox` I get the segmentation fault. I really don't know what I am doing wrong. I have tried exFAT, FAT32 but nothing works.

Also, every file in `cgi-bin` is 404, but I can open and download any file on sdcard root directory. So it looks like some kind of a problrm with httpd?

Hardware id is: V380E4_C3

Can I do anything else?

**Comment by Arkady23** [comment]
It's hard to say. Does ftp work? Alternatively, you can change the sd card, disable ftp, so there is more memory.

**Comment by h200101** [comment]
Hi!
I have tried disabling the ftp, but when I run the busybox I still get segmentation fault.

The `top` command:
```
Mem: 32288K used, 2412K free, 1584K shrd, 3644K buff, 10244K cached
CPU:  8.0% usr  9.3% sys  0.0% nic 82.3% idle  0.0% io  0.0% irq  0.2% sirq
Load average: 1.02 0.55 0.22 1/111 792```

Is there a way, I can enable rtsp via terminal?

**Comment by h200101** [comment]
Guys!!!

I figured it out!

Only thing I had to do, was go to `/mnt/mtd/mvconf/factory_const.ini`

and at the bottom of the file I have added 
```
[CONST_PARAM]
rtsp_enable=1 
```

and did a reboot. ONVIF and RTSP are now working perfectly! It also shows the option to enable or disable in the app!
